src/js/compat.ts

```diff
@@ -219,3 +276,89 @@ async function nodeLoadScript(url: string) {
     await import(/* webpackIgnore: true */ nodeUrlMod.pathToFileURL(url).href);
   }
 }
+
+/**
+ * Load a text file and executes it as Javascript
+ * @param url The path to load.
+ * @private
+ */
+async function greasemonkeyLoadScript(url: string) {
+  // If it's a url, load it with fetch then eval it.
+  //
+  // Check to see if we're loading pyodide.asm.js here
+  // cache asm.data so we can use emscripten's Module.getPreloadedPackage.
+  let urlParts = new URL(url);
+  let pathParts = urlParts.pathname.split("/");
+  let filename = pathParts[pathParts.length - 1];
+  if (filename == "pyodide.asm.js") {
+    let dataUrl = new URL(url);
+    let dataPathParts = dataUrl.pathname.split("/");
+    dataPathParts[dataPathParts.length - 1] = "pyodide.asm.data";
+    let dataPath = dataPathParts.join("/");
+    dataUrl.pathname = dataPath;
+    globalThis.asmData = await greasemonkey_loadBinaryFile(
+      dataUrl.href,
+      undefined,
+    );
+  }
+  // @ts-ignore
+  let response = await greasemonkey_get(url, "blob");
+  eval(response.responseText);
+}
+
+/**
+ * Load a file, only for use in Greasemonkey. Resolves relative paths against
+ * indexURL.
+ *
+ * @param path the path to load
+ * @returns a string containing the data
+ * @private
+ */
+export async function greasemonkey_loadFile(path: string): Promise<string> {
+  let response = await greasemonkey_get(path, "text");
+  return response.responseText;
+}
+
+export function greasemonkey_getPreloadedPackage(
+  remotePackageName: string,
+  remotePackageSize: number,
+): ArrayBuffer {
+  return globalThis.asmData.buffer;
+}
+
+/** @private */
+export let getPreloadedPackage: (
+  remotePackageName: string,
+  remotePackageSize: number,
+) => ArrayBuffer;
+if (IN_GM) {
+  getPreloadedPackage = greasemonkey_getPreloadedPackage;
+}
+
+export async function greasemonkey_instantiateWasm(
+  imports: object,
+  successCallback: any,
+): Promise<any> {
+  // @ts-ignore
+  let response = await greasemonkey_get(
+    this.locateFile("pyodide.asm.wasm"),
+    "arraybuffer",
```

Comment:
I need to figure out how to get rid of `this` here.  Maybe I'll move `locateFile` into compat.js?

Reply:
I don't fully understand what you are saying, but you might try `Module.locateFile`.

Reply:
Tried that... Module is only available in `pyodide.ts` not `compat.js`

Reply:
I'm getting ]a transpiler error](https://github.com/pyodide/pyodide/actions/runs/4236309322/jobs/7360979682#step:9:79) for this which is why I brought it up.  I can just add `@ts-ignore` if folks are ok with that.

Reply:
Added @ts-ignore: https://github.com/pyodide/pyodide/pull/3597/commits/147d4bdcc8f047640afff2b23b6b4a1b4085cc05#diff-0503c45203a13549426fcea74188746dd314feb61fe75c458485132d3de229b8R344